### ISSUE TITLE
Refactor to make capistrano optional

### ISSUE THREAD

**Issue by hackartisan** [curationexperts/ansible-hydra#15]
This will support the ability to build dev machines, both remote and in vagrant.

I'm thinking the best way to do this is to split up roles where necessary, possibly using some nested roles in places (e.g where the split is install vs. configure).

I considered using tags but they cannot be filtered from within a playbook (only from the command line) and I want to be able to encapsulate execution to a set of high-level playbooks without requiring a user to know a bunch of different command-line switches (which are easy to accidentally leave out).

**Comment by acozine** [comment]
I'd like to get to where ansible handles code deployment as well as infrastructure deployment. This looks promising: http://andrey.chernih.me/2015/03/15/deploying-ruby-on-rails-applications-with-ansible/

**Comment by hackartisan** [comment]
I like that idea too but I really need to break out capistrano or any other deployment-specific directory creation so that I can build dev machines.

I have this heavily in-progress and hope to submit a PR soon.

If you take on the capistrano replacement at some point, we'll need to make sure it does not break the ability to build usable dev instances, e.g. a vagrant box with code in a shared drive. Another potential conflict is all the shared directory stuff in capistrano.

**Comment by hackartisan** [comment]
Here's a WIP / first pass if you'd like to comment!

https://github.com/acozine/sufia-ansible/compare/master...chemheritage:cap-refactor

**Comment by hackartisan** [comment]
I've updated my cap-refactor branch; it's been tested and debugged so it builds a working production server on aws. Next step is to create a vagrant_dev playbook.

Also fixed a bug in ssl conf stuff. in a perfect world I would rebase that onto master and submit it now. Let me know if that is super important to you, @acozine

**Comment by acozine** [comment]
Thanks @hackmastera. @mark-dce has also been working on a fork of the ubuntu ansible repo, and has a a vagrant dev environment build he's happy with. I also have a working dev build (with a different approach) on the centos version. I'd like to check in about goals for the ansible project overall and whether DCE and CHF want to share a single ansible repo going forward. I'll send email about scheduling.

**Comment by hackartisan** [comment]
Srsly? @mark-dce can I look at what you've done?

**Comment by hackartisan** [comment]
Yeah, it looks like we're working in parallel. A call to check in sounds like a great idea.

**Comment by hackartisan** [comment]
This appears to have worked really well! Just pushed a few new commits to my branch; added the playbook to build a development-oriented vagrant, which still uses tomcat and apache, but in development mode. It also takes advantage of the shared directory so that a user can code on their local machine.

Also added a little bash script that lists all default vars in all roles.

I'd like to submit a PR or multiple PRs but I should split a couple things off.

Initial questions:
- Should I squash commits before submitting the PR? If so I will definitely extract unrelated commits, like the ssl bug fix and the util script.
- Should I submit the refactor as its own PR, and then the new vagrant_dev playbook w/ instructions separately?

**Comment by mark-dce** [comment]
closed via #21
